### PR TITLE
[docker] quick script to copy binary

### PR DESCRIPTION
**Pull request by rustielin**
### Description

To iterate fast, and if you build binary against linux, you can copy the binary directly into a source image

Also miscellaneous cleanup of the main Dockerfile, to speed up a tiny bit

### Test Plan

Run it

### CODE REVIEW

File: docker/build-test-node.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    exit 1
+fi
+
+tempdir=$(mktemp -d)
+echo "Using tempdir $tempdir"
+cp $SOURCE_APTOS_NODE_BINARY $tempdir # copy the binary into docker dir for context
+cat <<EOF >$tempdir/Dockerfile
+FROM $SOURCE_IMAGE
+# Override the aptos-node binary at its installed location
+COPY $(basename $SOURCE_APTOS_NODE_BINARY) /usr/local/bin/aptos-node
```
Comment (Markuze) [inline docker/build-test-node.sh:28]: COPY works only if the file is in the docker context. You need to copy BINARY to tempdir

**Review by Markuze — CHANGES_REQUESTED** [review]
Please fix a minor issue in the build script.

File: docker/build-test-node.sh (hunk 1 above)
Comment (rustielin) [reply docker/build-test-node.sh:28]: Yea I fixed this with the latest update. See the following on line 24 and 28:

```
cp $SOURCE_APTOS_NODE_BINARY $tempdir # copy the binary into docker dir for context
...
COPY $(basename $SOURCE_APTOS_NODE_BINARY) /usr/local/bin/aptos-node
```

**Review by Markuze — APPROVED** [review]

File: docker/build-test-node.sh (hunk 1 above)
Comment (Markuze) [reply docker/build-test-node.sh:28]: We still have the libssl issue, if you can solve this script will be invaluable.

**Pull request closed without merging**